pkg/operator/configobservation/cloudprovider/observe_cloudprovider.go

```diff
@@ -27,71 +22,53 @@ func ObserveCloudProviderNames(genericListers configobserver.Listers, recorder e
 	var errs []error
 	cloudProvidersPath := []string{"extendedArguments", "cloud-provider"}
 
-	previouslyObservedConfig := map[string]interface{}{}
-	if currentCloudProvider, _, _ := unstructured.NestedStringSlice(existingConfig, cloudProvidersPath...); len(currentCloudProvider) > 0 {
-		if err := unstructured.SetNestedStringSlice(previouslyObservedConfig, currentCloudProvider, cloudProvidersPath...); err != nil {
-			errs = append(errs, err)
+	prevObservedConfig := map[string]interface{}{}
+	currentCloudProvider, _, err := unstructured.NestedStringSlice(existingConfig, cloudProvidersPath...)
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
+	if len(currentCloudProvider) > 0 {
+		if err := unstructured.SetNestedStringSlice(prevObservedConfig, currentCloudProvider, cloudProvidersPath...); err != nil {
+			return prevObservedConfig, append(errs, err)
 		}
 	}
 
+	if !listers.InfrastructureSynced() {
```

Comment:
remove.  We no longer need this because of cluster-config-operator

Reply:
@soltysh https://github.com/soltysh/cluster-kube-controller-manager-operator/pull/1

Reply:
Thanks Luis :heart: